br/pkg/utils/memory_monitor.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
```

Comment:
Is is possible to reuse the infras in `pkg/util/memoryusagealarm/memoryusagealarm.go`?